mh/session.py

```diff
@@ -162,6 +162,8 @@ def layer_start(self):
     def layer_end(self):
         if self.layer > 1:
             # City path
+            if self.local_info['circle_id'] is not None:
```

Comment:
Ditto
```python
if self.local_info['circle_id']:
```

Reply:
it can be zero due to python indexing, we add 1 to it before we send it off

Reply:
Done